src/Core/Managed/Operation.CC.Shared/Extensibility/Implementation/CallContextBasedOperationCorrelationTelemetryInitializer.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation
+{
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Extensibility;
+
+    /// <summary>
+    /// Telemetry initializer that populates OperationContext for the telemetry item based on context stored in CallContext.
+    /// </summary>
+    internal class CallContextBasedOperationCorrelationTelemetryInitializer : ITelemetryInitializer
```

Comment:
I have a public extension method to add this telemetry initializer. It's good feedback however and I'll fix it in the next PR